### PR TITLE
Fix unreachable code reported by PHPstan

### PR DESCRIPTION
**Pull request by AdrienClairembault**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have performed a self-review of my code.

## Description

Investigate unreachable code reported by PHPstan.
I've fixed the easy cases and left some comment for special cases that will need more investigation.

In some switches, I've told phpstan to ignore the error in order to keep some unreachable code as a safety mechanism, let me know if you prefer not to (phpcs is already supposed to detect involuntary switch fall-through so this might be overkill).

One error will be fixed by #17806, I'll rebase once merged.

### CODE REVIEW

File: src/Webhook.php
Comment (AdrienClairembault) [inline src/Webhook.php:1]: Please disable white-space changes to better review this file.

**Comment by trasher** [comment]
Tests are failing

File: src/Notification_NotificationTemplate.php [hunk 1]
```diff
@@ -500,10 +500,7 @@ public static function getModeClass($mode, $extratype = '')
         } else if ($extratype === 'setting') {
             $classname = 'Notification' . ucfirst($mode) . 'Setting';
         } else {
-            if ($extratype !== '') {
-                throw new \LogicException(sprintf('Unknown type `%s`.', $extratype));
-            }
-            $classname = 'Notification' . ucfirst($mode);
+            throw new \LogicException(sprintf('Unknown type `%s`.', $extratype));
         }
```
Comment (cedric-anne) [inline src/Notification_NotificationTemplate.php:507]: This change is not correct. `Notification_NotificationTemplate::getModeClass('mailing')` is a valid call, and the expected returned class is `NotificationMailing`.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Comment by AdrienClairembault** [comment]
Rebased to main + fixed getModeClass issues.

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**